### DISCUSSION TITLE
Tree control component

### DISCUSSION THREAD

**Discussion by pchemguy** [sqlpage/SQLPage#408] [Ideas]
It would be nice to have a tree control component. And related to it, some kind of mechanism for dynamic data loading.

**Comment by lovasoa** [comment]
Yes, that would be nice ! What would be your use case for it ?

**Reply by pchemguy** [reply]
I use hierarchical categories to organize items, those are absolutely essential. I have about 4k items in my Zotero library, and it would be intractable without those. My parts library is smaller for now and I do not use hierarchical categories directly on them (well, I use them in Zotero, and I would probably bring those).

**Reply by lovasoa** [reply]
For these kinds of use cases, it is common to use the list component, and multiple pages. The list items in each page link to the next page.

**Reply by pchemguy** [reply]
What is your background? Did you have to do extensive literature research? I do not think you really understand what you are talking about here. Hierarchical categories is an essential information retrieval tool, and you can never replace it with flat lists. I mean, you can, of course, use flat lists, but they do not help much. Those are necessary when you do initial information search on search engines, and it costs a lot of time and effort. You really do not want to do the same inefficient time-consuming process over and over again...

Imagine you have four thousand items and you need to find, say, all database related resources and, separately, those focused on particular engine or function. One can stare at this countless pages all the time they want. You would be spending all your time on staring at these pages instead of doing the job. Same goes about emails, files. and so on. I use hierarchical categories everywhere they are available, and I usually do not even bother learning software designed for managing information that does not have this feature. I cannot stress enough how absolutely essential this feature is for anyone having to deal with substantial volumes of structured and semi-structured information, such as literature references, communications with vendors, complex equipment design files and so on (and you need those for any research - I left academia almost 10 yeas ago).

**Reply by lovasoa** [reply]
I meant that it is common *in SQLPage*. 

You generally have a first page with a list of first level items, and each one links to a list of second level items, and so on.

**Reply by pchemguy** [reply]
I see. Well, this won't work for me.
